pkg/updater/resultstore/resultstore.go

```diff
@@ -103,38 +105,76 @@ func ResultStoreColumnReader(client *DownloadClient, reprocess time.Duration) up
 			results = append(results, result)
 		}
 
-		processedResults := processRawResults(log, results)
+		invocations := processRawResults(log, results)
 
 		// Reverse-sort invocations by start time.
-		sort.SliceStable(processedResults, func(i, j int) bool {
-			return processedResults[i].InvocationProto.GetTiming().GetStartTime().GetSeconds() > processedResults[j].InvocationProto.GetTiming().GetStartTime().GetSeconds()
+		sort.SliceStable(invocations, func(i, j int) bool {
+			return invocations[i].InvocationProto.GetTiming().GetStartTime().GetSeconds() > invocations[j].InvocationProto.GetTiming().GetStartTime().GetSeconds()
 		})
 
-		for _, pr := range processedResults {
-			inflatedCol := processGroup(tg, pr)
+		for _, inv := range invocations {
+			inflatedCol := processGroup(tg, inv)
 			receivers <- *inflatedCol
 		}
 		return nil
 	}
 }
 
-func processRawResults(log logrus.FieldLogger, results []*fetchResult) []*processedResult {
-	var processedResults []*processedResult
+// invocationGroup will contain info on the groupId and all invocations for that group
+// a group will correspond to a column after transformation
+type invocationGroup struct {
+	GroupId     string
+	Invocations []*invocation
+}
+
+// groupInvocations will group the invocations according to the grouping strategy in the config
+// [inv1,inv2,inv3,inv4] -> [[inv1,inv2,inv3], [inv4]]
+func groupInvocations(log logrus.FieldLogger, tg *configpb.TestGroup, invocations []*invocation) []*invocationGroup {
+	groupedInvocations := make(map[string]*invocationGroup)
+
+	var sortedGroups []*invocationGroup
+
+	for _, invocation := range invocations {
+		groupIdentifier := extractGroupID(tg, invocation)
+		group, ok := groupedInvocations[groupIdentifier]
+		if !ok {
+			group = &invocationGroup{
+				GroupId: groupIdentifier,
+			}
+			groupedInvocations[groupIdentifier] = group
+		}
+		group.Invocations = append(group.Invocations, invocation)
+	}
+
+	for _, group := range groupedInvocations {
+		sortedGroups = append(sortedGroups, group)
+	}
+
+	// reverse sort groups by invocation time
```

Comment:
nit: I think sorting in the function is a bit opaque (especially if the function doc doesn't clarify it's sorting). imo sorting should happen more local to where it's needed, or this function should clarify it sorts the invocations.

Reply:
Added a comment to the function